bgp/modules/terms.py

```diff
@@ -113,3 +115,45 @@ def validate_isbn(isbn):
     def __init__(self):
         super(IsbnExtractorModule, self).__init__(self.validate_isbn)
 
+
+class PageTypeProcessor(object):
+
+    def __init__(self, modules,keyword):
+        """
+        :param modules: a dict of {'name': module}
+        """
+        self.modules = modules
+        self.keyword = keyword
+    
+    def run(self, book):
+        self.pageNo = self.search(book.xml,keyword = self.keyword)
+
+    @classmethod
+    def search(cls,xmlFile,keyword = ""):
+        utf8_parser = etree.XMLParser(encoding='utf-8')
+        node = etree.fromstring(xmlFile.encode('utf-8'), parser=utf8_parser)
+        for x in node.iter('OBJECT'):
+            # Iterating over each WORD tag to locate the keyword
+            for i in x.iter('WORD'):
+                '''
+                Uncomment the below line to see all the keywords present in DJVU file
+                '''
+                #print(i.text)
+                if(i.text == keyword):
+                    '''
+                        x[0].attrib['value'] is Descartes-TheGeometry_0273.djvu 
+                        splitting at '.' and finding the last 4 lettes of the book name in this case it is 0273
+                    '''
+                    param = x[0].attrib['value'].split('.')[0] 
+                    pageNo = param[-4:]
+                    print("Keyword",keyword, "is found at page",pageNo) # For debugging purpose
+
+
+    @property
+    def results(self):
+        return self.pageNo
+
+class KeywordPageDetectorModule(PageTypeProcessor):
```

Comment:
The appropriate name would be PageDetectorModule and shouldn't the inheritance be reversed as in PageTypeProcessor should inherit KeywordPageDetectorModule and not the other way round?

Reply:
I think this is correct; The KeywordPageDetectorModule inherits/extends the capabilities and interfaces of PageTypeProcessor.